src/dreyfus/src/dreyfus_rpc.erl

```diff
@@ -102,10 +102,12 @@ disk_size(DbName, DDoc, IndexName) ->
     end.
 
 get_or_create_db(DbName, Options) ->
+    DbOpts = mem3_shards:opts_for_db(DbName),
```

Comment:
I don't have a very strong opinion against this approach. Just wanted to mention that it looks like leaking abstraction where all components need to deal with internal details of another app. Alternative solution could be:

mem3_util.erl
```
get_or_create_db(DbName, Options, CreateFun) ->
    DbOpts = mem3_shards:opts_for_db(DbName),
    Options1 = mem3_util:merge_opts(DbOpts, Options),
    case couch_db:open_int(DbName, Options1) of           
         {not_found, no_db_file} ->
                couch_log:warning("~p creating ~s", [?MODULE,  DbName]),
                CreateFun(DbName, Options1);
          Else ->
                Else
end
```

dreyfus_rpc.erl (and hastings_rpc.erl)
```
get_or_create_db(DbName, Options) ->
    mem3_util:get_or_create_db(DbName, Options, fun couch_server:create/2).
```

Reply:
I lean a bit more towards keeping the abstraction intact as much as possible

Reply:
Yeah... I wasn't a huge fan of exposing `mem3_util:merge_opts`, but it was the simplest thing to get this working. I'm not a big fan of passing in a create function here, I don't think we should expose that level of abstraction for raw database creation without an explicit need and desire for plugging in additional creation options, which I don't think we have any at the moment. Let me see if I can clean this up a bit.